bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+payments via this channel (i.e. be part of the public network); requiring their
+aggregated MuSig2 Schnorr signature proves that they control the channel.
+
+The blacklisting of conflicting nodes disallows multiple different
+announcements. Such conflicting announcements should never be broadcast by any
+node, as this implies that keys have leaked.
+
+While channels should not be advertised before they are sufficiently deep, the
+requirement against rebroadcasting only applies if the transaction has not moved
+to a different block.
+
+In order to avoid storing excessively large messages, yet still allow for
+reasonable future expansion, nodes are permitted to restrict rebroadcasting
+(perhaps statistically).
+
+New channel features are possible in the future: backwards compatible (or
+optional) features will have _odd_ feature bits, while incompatible features
+will have _even_ feature bits
+(["It's OK to be odd!"](00-introduction.md#glossary-and-terminology-guide)).
+
+A delay of 12-blocks is used when forgetting a channel on funding output spend
+as to permit a new `channel_announcement_2` to propagate which indicates this
+channel was spliced.
+
+## The `node_announcement_2` Message
+
+TODO: complete
+
+1. type: xxx (`node_announcement_2`)
+2. data: (tlv_stream):
+    1. type: 0 (`announcement_sig`)
+    2. data:
+        * [`schnorr_signature`:`schnorr_signature`]
+    3. type: 1 (`features`)
+    4. data:
+        * [`...*byte`: `features`]
+    5. type: 2 (`timestamp`)
+    6. data:
+        * [`u32`: `block_height`]
+    7. type: 4 (`color`)
+    8. data:
+        * [`3*byte`:`rgb_color`]
+    9. type: 5 (`alias`)
+    10. data:
+         * [`32*byte`:`alias`]
+    11. type: 6 (`ipv4_addrs`)
+    12. data: 
+         * [`...*byte`: `ipv4_addresses`]  
+    13. type: 7 (`ipv6_addrs`)
+    14. data:
+         * [`...*byte`: `ipv6_addresses`]
+    15. type: 8 (`tor_v3_addrs`)
+    16. data: 
+         * [`...*byte`: `tor_v3_addrs`]
+    17. type: 9 (`dns_hostname`)
+    18. data:
+         * [`...*byte`: `dns_hostname`]
+
+## The `channel_update_2` Message
+
+TODO: complete
+
+1. type: xxx (`channel_update_2`)
+2. data: (tlv_stream):
+    1. type: 0 (`update_sig`)
+    2. data:
+        * [`schnorr_signature`:`schnorr_signature`]
+    3. type: 2 (`chain_hash`)
+    4. data:
+        * [`chain_hash`:`chain_hash`]
+    5. type: 2 (`timestamp`)
+    6. data:
+        * [`u32`: `block_height`]
+    7. type: 3 (`sequence`)
+    8. data:
+        * [`u8`, `sequence`]
+    9. type: 4 (`message_flags`)
+    10. data:
+        * [`...*byte`, `message_flags`]
+    11. type: 5 (`channel_flags`)
+    12. data:
+        * [`...*byte`, `channel_flags`]
+    13. type: 6 (`cltv_expiry_delta`)
+    14. data:
+        * [`...*byte`, `cltv_expiry_delta`]
+    15. type: 7 (`htlc_minimum_msat`)
+    16. type: 8 (`fee_base_msat`)
+    17. type: 9 (`fee_prop_millionths`)
+    18. type: 10 (`htlc_max_msat`)
```

Comment:
Let's renumber to make min and max adjacent, and decide on max vs maximum and min vs minimum to keep pedants like me happy!

They're all tu64, ofc.